src/sdam/topology.ts

```diff
@@ -57,6 +57,7 @@ import { DestroyOptions, Connection } from '../cmap/connection';
 import { RunCommandOperation } from '../operations/run_command';
 import type { CursorOptions } from '../cursor/cursor';
 import type { MongoClientOptions } from '../mongo_client';
+import { Db, Collection, MongoClient } from '..';
```

Comment:
```suggestion
import { Db } from '../db';
import { Collection } from '../collection';
import { MongoClient } from '../mongo_client';
```
Just an 🧅  I think internally we should import from the file directly

Reply:
+1, also you may run into issues with require "cycles" with this approach (noted below).